File: templates/go-service/service/component/monitoring/component.go
package monitoring

import (
	"app/config"
	"app/core"
	"net/http"

	"github.com/gin-gonic/gin"
)

// Component component definition
type Component struct {
	config *config.Config
	app    core.App
	router *gin.Engine
}

// CreateComponent create new component
func CreateComponent(config *config.Config, app core.App, router *gin.Engine) *Component {
	return &Component{
		config: config,
		app:    app,
		router: router,
	}
}

// Setup component
func (comp *Component) Setup() {
	serviceName := comp.config.ServiceName

	comp.router.GET("/liveness", func(c *gin.Context) {
		liveness := comp.app.Liveness()
		// send response
		c.JSON(comp.getHTTPCodeByStatus(liveness), gin.H{
			"service_name": serviceName,
			"is_alive":     liveness,
		})
	})

	comp.router.GET("/readiness", func(c *gin.Context) {
		readiness := comp.app.Readiness()
		// send response
		c.JSON(comp.getHTTPCodeByStatus(readiness), gin.H{
			"service_name": serviceName,
			"is_ready":     readiness,
		})
	})

}

func (comp *Component) getHTTPCodeByStatus(status bool) (untyped int) {
	if status {
		return http.StatusOK
	}
	return http.StatusInternalServerError
}

File: templates/nodejs-service/service/src/components/monitoring/component.ts
import { Express } from "express";
import { App, Comp } from "../../core";
import { Config } from "../../config";

export class Component implements Comp {
    constructor(private config: Config, private app: App, private router: Express) { }

    setup() {
        const serviceName = this.config.serviceName;

        this.router.get("/liveness", (_, res) => {
            const liveness = this.app.liveness();
            const httpCode = liveness ? 200 : 500;
            res.status(httpCode).send({
                service_name: serviceName,
                is_alive: liveness,
            });
        });

        this.router.get("/readiness", (_, res) => {
            const readiness = this.app.readiness();
            const httpCode = readiness ? 200 : 500;
            res.status(httpCode).send({
                service_name: serviceName,
                is_ready: readiness,
            });
        });
    }
}